File: 2021-remove-all-occurrences-of-a-substring/remove-all-occurrences-of-a-substring.js
/**
 * @param {string} s
 * @param {string} part
 * @return {string}
 */
var removeOccurrences = function(s, part) {
    if(!s.includes(part)){
        return s;
    }
    s = s.replace(part, "")
    return removeOccurrences(s, part);
};

File: 2021-remove-all-occurrences-of-a-substring/remove-all-occurrences-of-a-substring.java
class Solution {
    public String removeOccurrences(String s, String part) {
        StringBuilder sb = new StringBuilder(s);
        while(sb.indexOf(part) != -1){
            int index = sb.indexOf(part);
            sb = sb.delete(index, index + part.length());
        }
        return sb.toString();
    }
}